sdk/client-core/core/src/main/java/io/clientcore/core/implementation/util/Slf4jLoggerShim.java

```diff
@@ -0,0 +1,242 @@
+            isWarnEnabled = isSlf4jEnabledAtLevel(slf4jLogger, WARNING) || defaultLogger.isEnabled(WARNING);
+            isErrorEnabled = isSlf4jEnabledAtLevel(slf4jLogger, ERROR) || defaultLogger.isEnabled(ERROR);
+        } catch (ReflectiveOperationException e) {
+            slf4jLogger = null;
+
+            DEFAULT_LOGGER.log(WARNING, "Failed to initialize Slf4jLoggerShim.", e);
+
+            isVerboseEnabled = defaultLogger.isEnabled(VERBOSE);
+            isInfoEnabled = defaultLogger.isEnabled(INFORMATIONAL);
+            isWarnEnabled = defaultLogger.isEnabled(WARNING);
+            isErrorEnabled = defaultLogger.isEnabled(ERROR);
+        }
+    }
+
+    public boolean canLogAtLevel(ClientLogger.LogLevel logLevel) {
+        if (logLevel == null) {
+            return false;
+        }
+
+        switch (logLevel) {
+            case VERBOSE:
+                return isVerboseEnabled;
+            case INFORMATIONAL:
+                return isInfoEnabled;
+            case WARNING:
+                return isWarnEnabled;
+            case ERROR:
+                return isErrorEnabled;
+            default:
+                return false;
+        }
+    }
+
+    public void performLogging(ClientLogger.LogLevel logLevel, String message, Throwable throwable) {
+        if (!canLogAtLevel(logLevel)) {
+            return;
+        }
+
+        if (slf4jLogger == null) {
+            // We've already included the exception stacktrace in the message, so there's no need to pass it again to
+            // the default logger. We'll still pass it to the SLF4J logger in case the provider wants to do something
+            // else with it.
+            defaultLogger.log(logLevel, message, null);
+
+            return;
+        }
+
+        try {
+            switch (logLevel) {
+                case VERBOSE:
+                    LOGGER_VERBOSE_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case INFORMATIONAL:
+                    LOGGER_INFO_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case WARNING:
+                    LOGGER_WARN_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                case ERROR:
+                    LOGGER_ERROR_METHOD_HANDLE.invoke(slf4jLogger, message, throwable);
+
+                    break;
+                default:
+                    // Don't do anything, this state shouldn't be possible.
+                    break;
+            }
+        } catch (Throwable e) {
+            defaultLogger.log(WARNING, "Failed to log message, SLF4J logging will be disabled.", e);
+
+            slf4jLogger = null;
+        }
+    }
+
+    private static Object createLogger(String className) {
+        if (LOGGER_FACTORY_GET_LOGGER_METHOD_HANDLE == null || NOP_LOGGER_CLASS == null) {
```

Comment:
Only need to check one of these as if SLF4J isn't reflectively supported both will be null

Reply:
this seems safer in case anything changes.